Geometry/src/LUXEGeometry.cpp

```diff
@@ -1,125 +1,112 @@
-                                true);
-            }
-        }
-    }
-
-    // Write GDML file
-    G4GDMLParser parser;
-    parser.SetOutputFileOverwrite(true);
-    parser.Write(gdmlPath, physWorld);
-
-    return std::make_tuple(physWorld, names, gdmlPath);
-}
+std::shared_ptr<Acts::Experimental::LayerStructureBuilder> makeLayerBuilder(std::string gdmlPath,
+                                                                            std::vector<std::string> names,
+                                                                            Acts::GeometryContext gctx,
+                                                                            Acts::ActsScalar zMin,
+                                                                            Acts::ActsScalar zMax) {
+    auto spCfg = Acts::Experimental::Geant4SurfaceProvider<3>::Config();
+    spCfg.gdmlPath = gdmlPath;
+    spCfg.surfacePreselector =
+            std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(names,
+                                                                                true);
+
+    auto kdtDOpt = Acts::Experimental::Geant4SurfaceProvider<3>::kdtOptions();
+    kdtDOpt.range = Acts::RangeXD<3, Acts::ActsScalar>();
+    kdtDOpt.range[0].set(zMin, zMax);
+    kdtDOpt.binningValues = {Acts::BinningValue::binZ};
 
+    auto SP = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<3>>(
+            spCfg, kdtDOpt, false);
+
+    auto lbCfg = Acts::Experimental::LayerStructureBuilder::Config();
+    lbCfg.surfacesProvider = SP;
+    auto LB =
+            std::make_shared<Acts::Experimental::LayerStructureBuilder>(lbCfg);
+    return LB;
+}
 std::shared_ptr<const Acts::Experimental::Detector> 
     buildLUXEDetector(std::string gdmlPath, 
         std::vector<std::string> names, 
         Acts::GeometryContext gctx) {
             // Default template parameters are fine
             // when using names as identifiers
-            auto spFullCfg = Acts::Experimental::Geant4SurfaceProvider<>::Config();
-            spFullCfg.gdmlPath = gdmlPath;
-            spFullCfg.surfacePreselector =
-                std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(names,
-                    true);
-    
-            auto spFull = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<>>(
-                spFullCfg, Acts::Experimental::Geant4SurfaceProvider<>::kdtOptions(),
-                false);
-    
-            auto lbFullCfg = Acts::Experimental::LayerStructureBuilder::Config();
-            lbFullCfg.surfacesProvider = spFull;
-    
-            auto lbFull =
-                std::make_shared<Acts::Experimental::LayerStructureBuilder>(lbFullCfg);
-    
-            auto [sFull, vFull, suFull, vuFull] = lbFull->construct(gctx);
-    
-            assert(sFull.size() == names.size());
+            auto layerBuilder1 = makeLayerBuilder(gdmlPath, names, gctx, 3942, 3972);
```

Comment:
Let's make the geo parameters some constants that we read out from a specialized namespace